api/atomicpool.go

```diff
@@ -0,0 +1,187 @@
+	ids := []common.TxID{}
+	for _, tx := range ag.Txs {
+		ids = append(ids, tx.TxID)
+	}
+	actualAGID := common.CalculateAtomicGroupID(ids)
+	return actualAGID == ag.ID
+}
+
+func (a *API) postAtomicPool(c *gin.Context) {
+	// Parse body
+	var receivedAtomicGroup AtomicGroup
+	if err := c.ShouldBindJSON(&receivedAtomicGroup); err != nil {
+		retBadReq(err, c)
+		return
+	}
+	// Validate atomic group id
+	if !receivedAtomicGroup.IsAtomicGroupIDValid() {
+		retBadReq(errors.New(ErrInvalidAtomicGroupID), c)
+		return
+	}
+	nTxs := len(receivedAtomicGroup.Txs)
+	if nTxs <= 1 {
+		retBadReq(errors.New(ErrSingleTxInAtomicEndpoint), c)
+		return
+	}
+	// Validate txs
+	txIDStrings := make([]string, nTxs) // used for successful response
+	clientIP := c.ClientIP()
+	for i, tx := range receivedAtomicGroup.Txs {
+		// Find requested transaction
+		relativePosition, err := requestOffset2RelativePosition(tx.RqOffset)
+		if err != nil {
+			retBadReq(err, c)
+			return
+		}
+		requestedPosition := i + relativePosition
+		if requestedPosition > len(receivedAtomicGroup.Txs)-1 || requestedPosition < 0 {
+			retBadReq(errors.New(ErrRqOffsetOutOfBounds), c)
+			return
+		}
+		// Set fields that are omitted in the JSON
+		requestedTx := receivedAtomicGroup.Txs[requestedPosition]
+		receivedAtomicGroup.Txs[i].RqFromIdx = requestedTx.FromIdx
+		receivedAtomicGroup.Txs[i].RqToIdx = requestedTx.ToIdx
+		receivedAtomicGroup.Txs[i].RqToEthAddr = requestedTx.ToEthAddr
+		receivedAtomicGroup.Txs[i].RqToBJJ = requestedTx.ToBJJ
+		receivedAtomicGroup.Txs[i].RqTokenID = requestedTx.TokenID
+		receivedAtomicGroup.Txs[i].RqAmount = requestedTx.Amount
+		receivedAtomicGroup.Txs[i].RqFee = requestedTx.Fee
+		receivedAtomicGroup.Txs[i].RqNonce = requestedTx.Nonce
+		receivedAtomicGroup.Txs[i].ClientIP = clientIP
+		receivedAtomicGroup.Txs[i].AtomicGroupID = receivedAtomicGroup.ID
+
+		// Validate transaction
+		if err := a.verifyPoolL2Tx(receivedAtomicGroup.Txs[i]); err != nil {
+			retBadReq(err, c)
+			return
+		}
+
+		// Prepare response
+		txIDStrings[i] = receivedAtomicGroup.Txs[i].TxID.String()
+	}
+
+	// Validate that all txs in the payload represent a single atomic group
+	if !isSingleAtomicGroup(receivedAtomicGroup.Txs) {
+		retBadReq(errors.New(ErrTxsNotAtomic), c)
+		return
+	}
+	// Insert to DB
+	if err := a.l2.AddAtomicTxsAPI(receivedAtomicGroup.Txs); err != nil {
+		retSQLErr(err, c)
+		return
+	}
+	// Return IDs of the added txs in the pool
+	c.JSON(http.StatusOK, txIDStrings)
+}
+
+// requestOffset2RelativePosition translates from 0 to 7 to protocol position
+func requestOffset2RelativePosition(rqoffset uint8) (int, error) {
+	const rqOffsetZero = 0
+	const rqOffsetOne = 1
+	const rqOffsetTwo = 2
+	const rqOffsetThree = 3
+	const rqOffsetFour = 4
+	const rqOffsetFive = 5
+	const rqOffsetSix = 6
+	const rqOffsetSeven = 7
+	const rqOffsetMinusFour = -4
+	const rqOffsetMinusThree = -3
+	const rqOffsetMinusTwo = -2
+	const rqOffsetMinusOne = -1
```

Comment:
```suggestion
	const (
	    rqOffsetMinusOne = -1
	    ...
	    rqOffsetMinusOne = -1
)	    
```

Reply:
Maybe is it better to use a map structure instead of a function with multiple ifs